[TS] types.map(types.map(types.map(types.boolean))) gives an error

When using
```
const map= types.map(types.boolean);
const mapOfMap= types.map(map);
const mapOfMapOfMap = types.map(mapOfMap);
```
TS (both 2.6.2 and 2.7.2) gives an error about mapOfMapOfMap:
```
 error TS2345: Argument of type 'IComplexType<{ [key: string]: any; }, IExtendedObservableMap<IExtendedObservableMap<boolean> & {}...' is not assignable to parameter of type 'IType<{ [key: string]: any; }, IExtendedObservableMap<IExtendedObservableMap<boolean> & {} & ISna...'.
  Types of property 'Type' are incompatible.
    Type 'IExtendedObservableMap<IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boo...' is not assignable to type 'IExtendedObservableMap<IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boo...'. Two different types with this name exist, but they are unrelated.
      Type 'IExtendedObservableMap<IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boo...' is not assignable to type '{}'.
        Types of property 'toJSON' are incompatible.
          Type '() => IKeyValueMap<IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boolean...' is not assignable to type '(() => { [key: string]: any; }) | undefined'.
            Type '() => IKeyValueMap<IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boolean...' is not assignable to type '() => { [key: string]: any; }'.
              Type 'IKeyValueMap<IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boolean; }>>' is not assignable to type '{ [key: string]: any; }'.
                Index signatures are incompatible.
                  Type 'IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boolean; }>' is not assignable to type '{ [key: string]: boolean; }'.
                    Index signature is missing in type 'IExtendedObservableMap<boolean> & {} & ISnapshottable<{ [key: string]: boolean; }>'.
```

funnily enough map and mapOfMap are ok

PS: this used to be ok before mbst 1.4

--- a/packages/mobx-state-tree/src/types/complex-types/map.ts
+++ b/packages/mobx-state-tree/src/types/complex-types/map.ts
@@ -294,7 +294,7 @@ export class MapType<S, T> extends ComplexType<{ [key: string]: S }, IExtendedOb
  */
 export function map<S, T>(
     subtype: IComplexType<S, T>
-): IComplexType<{ [key: string]: S }, IExtendedObservableMap<T>>;
+): IComplexType<{ [key: string]: S }, IExtendedObservableMap<T>>
 export function map<S, T>(
     subtype: IType<S, T>
 ): IComplexType<{ [key: string]: S }, IExtendedObservableMap<T>> {

--- a/packages/mobx-state-tree/src/types/complex-types/map.ts
+++ b/packages/mobx-state-tree/src/types/complex-types/map.ts
@@ -293,6 +293,9 @@ export class MapType<S, T> extends ComplexType<{ [key: string]: S }, IExtendedOb
  * @returns {IComplexType<S[], IObservableArray<T>>}
  */
 export function map<S, T>(
+    subtype: IComplexType<S, T>
+): IComplexType<{ [key: string]: S }, IExtendedObservableMap<T>>;
+export function map<S, T>(
     subtype: IType<S, T>
 ): IComplexType<{ [key: string]: S }, IExtendedObservableMap<T>> {
     return new MapType<S, T>(`map<string, ${subtype.name}>`, subtype)

